### PR TITLE
feat: Provision 32-bit GPU driver libraries

### LINKED ISSUE #2343: Provision 32-bit NVIDIA driver libraries with --nv
### Version of Apptainer

1.3.3

### Expected behavior

Some applications (including Wine - for running Windows `.exe` in Linux) require 32-bit NVIDIA driver libraries. The NVIDIA container toolkit supports this option using `NVIDIA_DRIVER_CAPABILITIES=compat32`.

### Actual behavior

Only 64-bit NVIDIA libraries are provisioned using `--nv`.

Therefore, it would benefit to add an (example) `x86` directory within `/.singularity.d/libs` and provision 32-bit libraries.

### What OS/distro are you running

Ubuntu 22.04.4

### How did you install Apptainer

PPA

### PR DESCRIPTION
**Pull request by ehfd**
Reviewer: @DrDaveD 

## Description of the Pull Request (PR):

Implements 32-bit driver library injection as `--compat32` on NVIDIA/ROCm setups.

More information in https://github.com/apptainer/apptainer/issues/2343.

### This fixes or addresses the following GitHub issues:

 - Fixes #2343

#### Before submitting a PR, make sure you have done the following:

- [x] Read the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] Make sure all commits are signed-off with `git commit -s`, see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [x] Added changes to the [CHANGELOG](https://github.com/apptainer/apptainer/blob/main/CHANGELOG.md), if necessary according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [x] Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)).
- [x] Based this PR against the appropriate branch according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [x] Added myself as a contributor to the [Contributors File](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTORS.md).

### CODE REVIEW

File: e2e/gpu/gpu.go [hunk 1]
```diff
@@ -107,6 +107,74 @@ func (c ctx) testNvidiaLegacy(t *testing.T) {
...
+	// Use Ubuntu 20.04 as this is a recent distro officially supported by Nvidia CUDA.
+	// We can't use our test image as it's alpine based and we need a compatible glibc.
+	imageURL := "docker://ubuntu:20.04"
+	imageFile, err := fs.MakeTmpFile("", "test-nvidia-compat32-", 0o755)
+	if err != nil {
+		t.Fatalf("Could not create test file: %v", err)
+	}
+	imageFile.Close()
+	imagePath := imageFile.Name()
+	defer os.Remove(imagePath)
```
Comment (DrDaveD) [inline e2e/gpu/gpu.go]: Switch to using `e2e.EnsureDebianImage()`.  That is an ubuntu version that is 20.04 or later, depending on the host version.  Currently the e2e host is 24.04 so it will use 24.04.

**Review by DrDaveD — CHANGES_REQUESTED** [review]
Other than the e2e change, this is looking pretty complete.

If you used an AI assist please be sure to follow the [AI Assistance Policy](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md#ai-assistance-policy).

**Comment by ehfd** [comment]
@DrDaveD Both done.

**Comment by ehfd** [comment]
@DrDaveD Hardened the e2e further, addressing a possible edge case where a node without 32-bit driver libraries might fail the test incorrectly.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,18 @@ For older changes see the [archived Singularity change log](https://github.com/a
 
 Changes since 1.5.x
 
+- Add a new `--compat32` action flag which, alongside `--nv` or `--rocm`,
```
Comment (DrDaveD) [inline CHANGELOG.md]: Please update the changelog entry to mention that this is a build flag in addition to an action flag.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (ehfd) [reply CHANGELOG.md]: Done.

Comment (DrDaveD) [reply CHANGELOG.md]: The new commit doesn't have a `Signed-off-by` so it needs to be redone.  Might as well redo them as one commit (with `Assisted-by`).  Anyway I think it's better to say at the beginning that it's an action and build flag than to add it to the end.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by ehfd** [comment]
@DrDaveD Fixed through rebasing.

**Review by DrDaveD — APPROVED** [review]
